src/main/java/net/porillo/objects/OffsetBounty.java

```diff
@@ -71,4 +92,220 @@ public int hashCode() {
+				if (offsetBounty.getHunterId() != null) {
+					UUID hunter = playerTable.getUuidMap().get(offsetBounty.getHunterId());
+					if (hunter != null) {
+						String hunterName = Bukkit.getOfflinePlayer(hunter).getName();
+						if (hunterName != null) {
+							row.add(hunterName);
+						}
+					}
+				}
+
+				//Or a link to join the bounty:
+				// - Track bounty IDs for click-events:
+				if (row.size() == 1) {
+					row.add(Lang.BOUNTY_JOIN.get());
+					clickIds.add(offsetBounty.getUniqueId());
+				}
+
+				//Remaining tree-blocks:
+				row.add(offsetBounty.getLogBlocksTarget().toString());
+
+				//Reward:
+				row.add(String.format("$%s", offsetBounty.getReward()));
+
+				//Add table row:
+				chatTable.addRow(row);
+			}
+
+			String json = chatTable.toJson(gPlayer, Lang.BOUNTY_JOIN.get(), "/gw bounty join", clickIds);
+			Bukkit.getServer().dispatchCommand(
+				Bukkit.getConsoleSender(),
+				json);
+		} catch (Exception e) {
+			gPlayer.sendMsg(Lang.BOUNTY_ERROR);
+			e.printStackTrace();
+		}
+	}
+
+	/**
+	 * Update the remaining blocks for an active bounty
+	 * - Requires a Vault-economy plugin to deposit funds
+	 */
+	public static OffsetBounty update(GPlayer hunter, int blocksCompleted) {
+		OffsetTable offsetTable = GlobalWarming.getInstance().getTableManager().getOffsetTable();
+		OffsetBounty bounty = offsetTable.update(hunter, blocksCompleted);
+		if (bounty != null && bounty.getTimeCompleted() != 0) {
+			Economy economy = GlobalWarming.getEconomy();
+			if (economy != null) {
+				economy.depositPlayer(hunter.getPlayer(), bounty.getReward());
+				notify(
+					bounty,
+					hunter,
+					String.format(Lang.BOUNTY_COMPLETEDBY.get(), hunter.getPlayer().getName()),
+					String.format(Lang.BOUNTY_COMPLETED.get(), bounty.getReward()));
+			}
+		}
+
+		return bounty;
+	}
+
+	/**
+	 * Join a bounty
+	 * - Players are limited to joining one bounty
+	 * - Bounties are limited to one hunter
+	 */
+	public static OffsetBounty join(GPlayer gPlayer, int bountyId) {
+		OffsetBounty bounty = null;
+		try {
+			OffsetTable offsetTable = GlobalWarming.getInstance().getTableManager().getOffsetTable();
+			bounty = offsetTable.join(gPlayer, bountyId);
+		} catch (Exception e) {
+			if (e.getMessage().length() > 0) {
+				gPlayer.sendMsg(e.getMessage());
+			} else {
+				e.printStackTrace();
+			}
+		}
+
+		return bounty;
+	}
+
+	public static OffsetBounty unJoin(GPlayer gPlayer) {
+		OffsetTable offsetTable = GlobalWarming.getInstance().getTableManager().getOffsetTable();
+		return offsetTable.unJoin(gPlayer);
+	}
+
+	public static void cancel(GPlayer gPlayer) {
+		int refund = 0;
+		OffsetTable offsetTable = GlobalWarming.getInstance().getTableManager().getOffsetTable();
+		List<OffsetBounty> cancelledBounties = offsetTable.cancel(gPlayer);
+		Economy economy = GlobalWarming.getEconomy();
+		if (economy != null) {
+		for (OffsetBounty bounty : cancelledBounties) {
+				economy.depositPlayer(gPlayer.getPlayer(), bounty.getReward());
```

Comment:
I know there is properly spam protection built in here, but we should consider implementing a refund penalty to send a price signal about making commitments and backing out. Once someone commits it should be a bit of a waste to rescind their carbon reductions offer.

Reply:
Players can only cancel idle bounties (if a player is actively hunting, the bounty won't be cancelled / refunded).  We can still charge a cancellation fee though.  Maybe some configurable percentage?

Reply:
Sure, the configurable percentage penalty will be nice but it's not strictly required at this time. More of a precaution thing from the viewpoint of a potential server owner.